build-system/tasks/runtime-test.js

```diff
@@ -541,96 +547,124 @@ function runTests() {
   const sectionMarker =
       (argv.saucelabs || argv.saucelabs_lite) ? 'saucelabs' : 'local';
 
-  let resolver;
-  const deferred = new Promise(resolverIn => {resolver = resolverIn;});
-  new Karma(c, function(exitCode) {
-    if (shouldCollapseSummary) {
-      console./* OK*/log('travis_fold:end:console_errors_' + sectionMarker);
-    }
-    server.emit('kill');
-    if (exitCode) {
-      log(
-          red('ERROR:'),
-          yellow('Karma test failed with exit code ' + exitCode));
+  // Run Sauce Labs tests in batches to avoid timeouts when connecting to the
+  // Sauce Labs environment.
+  if (argv.saucelabs || argv.saucelabs_lite) {
+    let batch = 1;
+    let startIndex = 0;
+    let endIndex = batchSize;
+    while (startIndex < endIndex) {
+      const configBatch = Object.assign({}, c);
+      configBatch.browsers = saucelabsBrowsers.slice(startIndex, endIndex);
+      log(green('Batch #' + batch + ': Running tests on ' +
+        configBatch.browsers.length + ' Sauce Labs browser(s)...'));
+      await createKarmaServer(configBatch);
+      startIndex = batch * batchSize;
+      batch++;
+      endIndex = Math.min(batch * batchSize, saucelabsBrowsers.length);
     }
-    if (argv.coverage) {
-      if (process.env.TRAVIS) {
-        const codecovCmd =
-            './node_modules/.bin/codecov --file=test/coverage/lcov.info';
-        let flags = '';
-        if (argv.unit) {
-          flags = ' --flags=unit_tests';
-        } else if (argv.integration) {
-          flags = ' --flags=integration_tests';
-        }
-        log(green('INFO: ') + 'Uploading code coverage report to ' +
-            cyan('https://codecov.io/gh/ampproject/amphtml') + ' by running ' +
-            cyan(codecovCmd + flags) + '...');
-        const output = getStdout(codecovCmd + flags);
-        const viewReportPrefix = 'View report at: ';
-        const viewReport = output.match(viewReportPrefix + '.*');
-        if (viewReport && viewReport.length > 0) {
-          log(green('INFO: ') + viewReportPrefix +
-              cyan(viewReport[0].replace(viewReportPrefix, '')));
+  } else {
+    await createKarmaServer(c);
+  }
+  server.emit('kill');
+  exitCtrlcHandler(handlerProcess);
+
+  /**
+   * Creates and starts karma server
+   *
+   * @param {!Object} configBatch
+   *
```

Comment:
For conciseness, delete blank lines in JSDoc. (I believe VSCode will automatically format JSDoc for you if you start off typing `/**` on a new line.)

Reply:
Removed.